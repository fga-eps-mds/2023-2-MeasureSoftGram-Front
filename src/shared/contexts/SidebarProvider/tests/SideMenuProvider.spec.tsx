import React, { ReactNode } from 'react';
import { act, render, renderHook } from '@testing-library/react';

import { SideMenuProvider, useSideMenuContext } from '@contexts/SidebarProvider/SideMenuProvider';

interface Props {
  children: ReactNode;
}

const Provider = ({ children }: Props) => <SideMenuProvider>{children}</SideMenuProvider>;

describe('SideMenuProvider', () => {
  it('should render the SideMenuProvider component with children', () => {
    const { container } = render(<div>children</div>, { wrapper: Provider });

    expect(container).toMatchSnapshot();
  });
});

describe('useSideMenuContext', () => {
  it('should return the SideMenuContext correct', () => {
    const { result } = renderHook(() => useSideMenuContext(), { wrapper: Provider });

    expect(result.current).toEqual({ isCollapsed: false, toggleCollapse: expect.any(Function) });
  });

  it('should return the SideMenuContext correct when isCollapsed is true', () => {
    const { result } = renderHook(() => useSideMenuContext(), { wrapper: Provider });

    act(() => {
      result.current.toggleCollapse();
    });

    expect(result.current).toEqual({ isCollapsed: true, toggleCollapse: expect.any(Function) });
  });
});
