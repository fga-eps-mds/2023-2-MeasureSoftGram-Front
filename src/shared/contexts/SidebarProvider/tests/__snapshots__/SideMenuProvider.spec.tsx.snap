// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuProvider should render the SideMenuProvider component with children 1`] = `
<div>
  <div>
    children
  </div>
</div>
`;
